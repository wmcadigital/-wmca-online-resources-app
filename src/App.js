import React, { useReducer, useMemo, useEffect } from 'react';
import { initialState, reducer, GlobalState, GlobalDispatch } from './store';
import data from './data.json';

import Header from './components/Header';
import SideBarFilters from './components/sidebar/SidebarFilters';
import Results from './components/results/Results';

function App() {
  const [store, dispatch] = useReducer(reducer, initialState);
  const dispatchContex = useMemo(() => ({ dispatch }), [dispatch]);
  const storeContex = useMemo(() => ({ store }), [store]);

  const fetchData = () => {
    return dispatch({
      type: 'setAllJobsAfterFetch',
      payload: data
    });
  };

  useEffect(() => {
    document.body.classList = 'bg-white app-blog';
    fetchData();
  }, []);
  
  return (
    <div className="App">
      <Header />
      <GlobalDispatch.Provider value={dispatchContex}>
        <GlobalState.Provider value={storeContex}>
          <div className="container-wide bg-white pad-30">
            <div className="pure-g justify-between">
              <div className="pure-u-1 pure-u-md-1-4">
                <SideBarFilters />
              </div>
              <div className="pure-u-1 pure-u-md-5-8" id="stories">
                <Results />
              </div>
            </div>
          </div>
        </GlobalState.Provider>
      </GlobalDispatch.Provider>
    </div>
  );
}

export default App;
