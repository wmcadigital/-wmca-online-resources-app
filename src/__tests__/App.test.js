import React from 'react';
import { render, cleanup } from '@testing-library/react';
import App from '../App';

afterEach(cleanup);

test('renders Find an opportunity', () => {
  const { getByText } = render(<App />);
  const linkElement = getByText(/Find an opportunity/i);
  expect(linkElement).toBeInTheDocument();
});
it('should take a snapshot', () => {
  const { asFragment } = render(<App />);
  expect(asFragment(<App />)).toMatchSnapshot();
});
