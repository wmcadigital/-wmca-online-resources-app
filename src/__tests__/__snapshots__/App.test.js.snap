// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should take a snapshot 1`] = `
<DocumentFragment>
  <div
    class="App"
  >
    <div
      class="wdgt-page-hero text-center"
      style="background-image: url(http://static.centro.org.uk/wmcaRedev/build/img/Peoples-stories-hero.jpg);"
    >
      <div
        class="splash"
      >
        <h1>
          Find an opportunity
        </h1>
      </div>
      <div
        class="wdgt-page-hero-overlay"
      />
    </div>
    <div
      class="container-wide bg-white pad-30"
    >
      <div
        class="pure-g justify-between"
      >
        <div
          class="pure-u-1 pure-u-md-1-4"
        >
          <div>
            <div
              class="container-wide bg-white"
            >
              <div
                class="pure-g justify-between"
              >
                <div
                  class="pure-u-1"
                >
                  <h2>
                    Filter
                  </h2>
                  <div
                    class="wmca-form wdgt"
                  >
                    <label
                      class="wmca-form-label filter-title"
                    >
                      Provider:
                    </label>
                    <div />
                    <div />
                    <div />
                    <div />
                    <div />
                    <div />
                    <div />
                    <div />
                    <div />
                    <div />
                    <div />
                    <div />
                    <div />
                    <div />
                    <div />
                    <div />
                    <div />
                    <div />
                    <div />
                    <div />
                    <div />
                    <div />
                    <div />
                    <div />
                    <div />
                    <div />
                    <div />
                    <div />
                    <div />
                    <div />
                    <div />
                    <div />
                    <div />
                    <div />
                    <div />
                    <div />
                    <div />
                    <div />
                    <div />
                    <div />
                    <div />
                    <div />
                    <div />
                    <div />
                    <div />
                    <div />
                    <div />
                    <div />
                  </div>
                  <div
                    class="wmca-form wdgt"
                  >
                    <label
                      class="wmca-form-label filter-title"
                    >
                      SupportFor:
                    </label>
                    <div />
                    <div />
                    <div />
                    <div />
                    <div />
                    <div />
                    <div />
                  </div>
                  <div
                    class="wmca-form wdgt"
                  >
                    <label
                      class="wmca-form-label filter-title"
                    >
                      Areas:
                    </label>
                    <div />
                    <div />
                    <div />
                  </div>
                  <div
                    class="wmca-form wdgt"
                  >
                    <label
                      class="wmca-form-label filter-title"
                    >
                      CourseArea:
                    </label>
                    <div />
                    <div />
                    <div />
                    <div />
                    <div />
                    <div />
                    <div />
                    <div />
                    <div />
                    <div />
                    <div />
                    <div />
                    <div />
                    <div />
                    <div />
                    <div />
                    <div />
                    <div />
                    <div />
                    <div />
                    <div />
                    <div />
                    <div />
                    <div />
                    <div />
                    <div />
                    <div />
                    <div />
                    <div />
                    <div />
                  </div>
                  <div
                    class="wmca-form wdgt"
                  >
                    <label
                      class="wmca-form-label filter-title"
                    >
                      Level:
                    </label>
                    <div />
                    <div />
                    <div />
                    <div />
                    <div />
                    <div />
                    <div />
                    <div />
                    <div />
                    <div />
                    <div />
                    <div />
                    <div />
                    <div />
                  </div>
                  <div
                    class="wmca-form wdgt"
                  >
                    <label
                      class="wmca-form-label filter-title"
                    >
                      Cost:
                    </label>
                    <div />
                    <div />
                    <div />
                    <div />
                    <div />
                    <div />
                    <div />
                    <div />
                    <div />
                    <div />
                    <div />
                    <div />
                    <div />
                    <div />
                    <div />
                    <div />
                    <div />
                    <div />
                    <div />
                  </div>
                  <div
                    class="wmca-form wdgt"
                  >
                    <label
                      class="wmca-form-label filter-title"
                    >
                      Type:
                    </label>
                    <div />
                    <div />
                    <div />
                    <div />
                  </div>
                </div>
              </div>
            </div>
          </div>
        </div>
        <div
          class="pure-u-1 pure-u-md-5-8"
          id="stories"
        >
          <div>
            Hello from results
          </div>
        </div>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
